aaelf64-morello/aaelf64-morello.rst

```diff
@@ -610,21 +648,29 @@ Relocations needed to define the traditional TLS models are undefined.
-    | 57602 | ``R_MORELLO_TLSDESC_CALL``       | None                       | For relaxation only. Must be used to identify a ``BLR``   |
-    |       |                                  |                            | instruction which performs an indirect call to the TLS    |
-    |       |                                  |                            | descriptor function for ``S + A``.                        |
-    +-------+----------------------------------+----------------------------+-----------------------------------------------------------+
+    +-------+-----------------------------------------+----------------------------+-----------------------------------------------------------+
+    | ELF64 | Name                                    | Operation                  | Comment                                                   |
+    | Code  |                                         |                            |                                                           |
+    +=======+=========================================+============================+===========================================================+
+    | 57600 | ``R_MORELLO_TLSDESC_ADR_PAGE20``        | ``Page(G(GTLSDESC(S+A)))`` | Set the immediate value of an ADRP to bits [31:12] of X.  |
+    |       |                                         |                            | Check that -2\ :sup:`31` <= X < 2\ :sup:`31`.             |
+    |       |                                         | ``- Page(P)``              |                                                           |
+    +-------+-----------------------------------------+----------------------------+-----------------------------------------------------------+
+    | 57601 | ``R_MORELLO_TLSDESC_LD128_LO12``        | ``G(GTLSDESC(S+A))``       | Set the LD/ST immediate field to bits [11:4] of X.        |
+    |       |                                         |                            | No overflow check. Check that X&15 = 0.                   |
+    +-------+-----------------------------------------+----------------------------+-----------------------------------------------------------+
+    | 57602 | ``R_MORELLO_TLSDESC_CALL``              | None                       | For relaxation only. Must be used to identify a ``BLR``   |
+    |       |                                         |                            | instruction which performs an indirect call to the TLS    |
+    |       |                                         |                            | descriptor function for ``S + A``.                        |
+    +-------+-----------------------------------------+----------------------------+-----------------------------------------------------------+
+    | 57603 | ``R_MORELLO_TLSIE_ADR_GOTTPREL_PAGE20`` | ``Page(G(GTPREL(S)))       | Set the immediate value of an ADRP to bits [31:12] of X.  |
```

Comment:
Is there a reason that the TLSIE relocations don't allow for an addend on the symbol? Just asking for my own understanding. Thanks

Reply:
Presumably because setting the size gets weird in that case; it needs to be done as an add after the scbnds

Reply:
I see, that makes sense, thanks